DataSetExplorer/Core/DataSets/Repository/DataSetInstanceDatabaseRepository.cs

```diff
@@ -20,6 +21,36 @@ public Instance GetDataSetInstance(int id)
             return _dbContext.DataSetInstances.Include(i => i.Annotations).FirstOrDefault(i => i.Id == id);
         }
 
+        public InstanceDTO GetInstanceWithRelatedInstances(int id)
+        {
+            var projects = _dbContext.DataSetProjects.Include(p => p.CandidateInstances)
+                .ThenInclude(c => c.Instances).ToList();
+            foreach (var project in projects)
```

Comment:
Not a fan of this. Is it possible to receive the Project ID from the front-end as well? The repository can return a complete project (including instances and related instances) and the service can pack the instance and project ID into a DTO.